ttnn/cpp/ttnn/tensor/types.cpp

```diff
@@ -293,3 +293,24 @@ MemoryConfig load_memory_config(const std::string& file_name) {
 }  // namespace tt_metal
 
 }  // namespace tt
+
+namespace ttnn {
+namespace types {
+
+const uint32_t Shape::operator[](std::int64_t index) const { return this->value.without_padding()[index]; }
+
+const uint32_t* Shape::begin() const {return this->value.without_padding().begin(); }
+const uint32_t* Shape::end() const {return this-> value.without_padding().end(); }
+
+tt::tt_metal::Array4D Shape::to_array_4D() const {
+    TT_ASSERT(this->rank() == 4, "Shape must be 4D for to_array_4D!");
```

Comment:
if its could be called by user it shoudl be TT_FATAL

Reply:
actually, i will be removing this function

Reply:
perfect, thanks!